src/states/common/approved.rs

```diff
@@ -154,36 +145,19 @@ pub trait Approved: Base {
             self.send_direct_message(p2p_node.peer_addr(), response);
         }
 
-        if poll {
-            self.parsec_poll(outbox)
-        } else {
-            Ok(Transition::Stay)
-        }
+        self.schedule_parsec_gossip();
     }
 
     fn handle_parsec_response(
         &mut self,
         msg_version: u64,
         par_response: parsec::Response,
         pub_id: PublicId,
-        outbox: &mut dyn EventBox,
-    ) -> Result<Transition, RoutingError> {
-        trace!(
-            "{} - handle parsec response v{} from {}",
-            self,
-            msg_version,
-            pub_id
-        );
-
+    ) {
         let log_ident = self.log_ident();
-        if self
-            .parsec_map_mut()
-            .handle_response(msg_version, par_response, pub_id, &log_ident)
-        {
-            self.parsec_poll(outbox)
-        } else {
-            Ok(Transition::Stay)
-        }
+        self.parsec_map_mut()
+            .handle_response(msg_version, par_response, pub_id, &log_ident);
+        self.schedule_parsec_gossip();
```

Comment:
So we are doing it both on response & request (on top of vote). So Start with 1 request will trigger 1 gossip from destination + 1 new gossip from source. i.e 1 -> 2 -> 4 -> 8 etc. So it seem very quickly everyone will gossip to everyone all the time. I don't see how the current schedule_parsec_gossip/send_scheduled_parsec_gossip would avoid this. Am I missing something?

Reply:
> Am I missing something?

Not necessarily. This is PR is very much work in progress and there are most likely issues. I just wanted to put it out so it can be tested already.

That said, perhaps we should not send that additional request on receiving request, only the response?